modules/core/src/main/scala/org/tessellation/infrastructure/snapshot/GlobalSnapshotStorage.scala

```diff
@@ -0,0 +1,66 @@
+package org.tessellation.infrastructure.snapshot
+
+import cats.Applicative
+import cats.data.NonEmptyList
+import cats.effect.{Async, Ref}
+import cats.syntax.applicativeError._
+import cats.syntax.flatMap._
+import cats.syntax.functor._
+import cats.syntax.order._
+
+import scala.util.control.NoStackTrace
+
+import org.tessellation.dag.snapshot.{GlobalSnapshot, SnapshotOrdinal, StateChannelSnapshotWrapper}
+import org.tessellation.domain.snapshot.GlobalSnapshotStorage
+import org.tessellation.ext.crypto._
+import org.tessellation.kryo.KryoSerializer
+import org.tessellation.schema.address._
+import org.tessellation.security.hash.Hash
+
+object GlobalSnapshotStorage {
+
+  def make[F[_]: Async: KryoSerializer](genesis: GlobalSnapshot): F[GlobalSnapshotStorage[F]] =
+    Ref[F].of[NonEmptyList[GlobalSnapshot]](NonEmptyList.of(genesis)).map(make(_))
+
+  def make[F[_]: Async: KryoSerializer](
+    snapshotsRef: Ref[F, NonEmptyList[GlobalSnapshot]]
+  ): GlobalSnapshotStorage[F] = new GlobalSnapshotStorage[F] {
+
+    def save(snapshot: GlobalSnapshot): F[Unit] =
+      snapshotsRef.modify { snapshots =>
+        val lastSnapshot = snapshots.head
+        lastSnapshot.hash match {
+          case Left(error) => (snapshots, error.raiseError[F, Unit])
+          case Right(lastSnapshotHash) =>
+            val expectedLink = (lastSnapshotHash, SnapshotOrdinal.next.next(lastSnapshot.ordinal))
```

Comment:
can you rebase and use `Next[SnapshotOrdinal]` syntax?

Reply:
I used the `_.next` syntax instead.